client/app/src/main/java/com/google/heartrate/androidos/app/gatt/GattCharacteristicManager.java

```diff
@@ -0,0 +1,54 @@
+package com.google.heartrate.androidos.app.gatt;
+
+import android.bluetooth.BluetoothGattCharacteristic;
+
+/**
+ * {@link GattCharacteristicManager} class provides method for safety exacting byte array data from
+ * different {@link BluetoothGattCharacteristic}.
+ */
+public class GattCharacteristicManager {
+
+    private GattCharacteristicManager() {}
+
+    /**
+     * Get value from characteristic.
+     *
+     * @param characteristic to get value from
+     * @return value characteristic
+     * @throws GattException if value can not be got
+     */
+    public static byte[] getValue(BluetoothGattCharacteristic characteristic) throws GattException {
+        try {
+            byte[] value = characteristic.getValue();
```

Comment:
You might want to check if the characteristic has the Read permission as this is a public method.
(same comment to similar functions)